### PR TITLE
various: pass logger.Logf through to more places

### PR DESCRIPTION
**Pull request by andrew-d**
Updates #7537

Change-Id: Id89acab70ea678c8c7ff0f44792d54c7223337c6

### CODE REVIEW

File: control/controlhttp/client.go [hunk 1]
```diff
@@ -382,18 +383,27 @@ func (a *Dialer) dialURL(ctx context.Context, u *url.URL, addr netip.Addr) (*Cli
 func (a *Dialer) tryURLUpgrade(ctx context.Context, u *url.URL, addr netip.Addr, init []byte) (net.Conn, error) {
 	var dns *dnscache.Resolver
 
+	// Maintain prior behaviour where passing no logger to
```
Comment (bradfitz) [inline control/controlhttp/client.go:386]: logf method instead?

File: tsnet/tsnet.go [hunk 2]
```diff
@@ -619,6 +620,25 @@ func (s *Server) logf(format string, a ...interface{}) {
 	log.Printf(format, a...)
 }
 
+// ReplaceGlobalLoggers will replace any Tailscale-specific package-global
+// loggers with this Server's logger. It returns a function that, when called,
+// will undo any changes made.
+//
+// Note that calling this function from multiple Servers will result in the
+// last call taking all logs; logs are not duplicated.
+func (s *Server) ReplaceGlobalLoggers() func() {
```
Comment (bradfitz) [inline tsnet/tsnet.go:629]: Name result?

**Review by bradfitz — APPROVED** [review]

File: control/controlhttp/client.go (hunk 1 above)
Comment (maisem) [inline control/controlhttp/client.go:386]: Maintain prior behavior from when?

File: net/dnsfallback/dnsfallback.go [hunk 3]
```diff
@@ -250,10 +250,11 @@ func SetCachePath(path string) {
 // logfunc stores the logging function to use for this package.
 var logfunc syncs.AtomicValue[logger.Logf]
 
-// SetLogger sets the logging function that this package will use. The default
-// logger if this function is not called is 'log.Printf'.
-func SetLogger(log logger.Logf) {
-	logfunc.Store(log)
+// SetLogger sets the logging function that this package will use, and returns
+// the old value (which may be nil). The default logger if this function is not
```
Comment (maisem) [inline net/dnsfallback/dnsfallback.go:254]: The default value of the logger is log.Printf

File: control/controlhttp/client.go (hunk 1 above)
Comment (andrew-d) [reply control/controlhttp/client.go:386]: I intentionally didn't use that; the `a.logf` method will drop logs if no `a.Logf` option is specified, whereas the previous behaviour of `dnscache.Resolver` always logs with `log.Printf` if there's no logger specified.

**Review by maisem — APPROVED** [review]

File: control/controlhttp/client.go (hunk 1 above)
Comment (bradfitz) [reply control/controlhttp/client.go:386]: logfOrDefault method then? I just don't want to see a big comment and var decl/conditional in the middle of unrelated code

File: net/dnsfallback/dnsfallback.go (hunk 3 above)
Comment (andrew-d) [reply net/dnsfallback/dnsfallback.go:254]: Yep, but the default value of the `logfunc` is nil; the top-level `logf` function will fall back to `log.Printf` if so.

Comment (maisem) [reply net/dnsfallback/dnsfallback.go:254]: The value defaults to log.Printf if not explicitly set?

File: control/controlhttp/client.go (hunk 1 above)
Comment (andrew-d) [reply control/controlhttp/client.go]: I actually just changed this to use `a.Logf` directly in the argument; this propagates nil-ness and retains the default "fall back to log.Printf" behaviour that `dnscache.Resolver` has. PTAL?

Comment (andrew-d) [reply control/controlhttp/client.go]: Changed the approach here to pass along `a.Logf` directly

**Review by maisem — APPROVED** [review]

**Pull request merged**